pkg/channel/distributed/controller/kafkachannel/controller.go

```diff
@@ -81,36 +80,25 @@ func NewController(ctx context.Context, cmw configmap.Watcher) *controller.Impl
 	// Get The K8S Client
 	kubeClientset := kubeclient.Get(ctx)
 
-	// Get & Validate The Kafka Auth Secret
-	kafkaSecret, err := util.GetKafkaSecret(ctx, kubeClientset, environment.SystemNamespace)
+	configuration, err := sarama.LoadEventingKafkaSettings(configMap)
 	if err != nil {
-		logger.Fatal("Failed To Load Kafka Auth Secret", zap.Error(err))
-	} else if kafkaSecret == nil {
-		logger.Fatal("No Kafka Auth Secret Found")
-	} else {
-		logger.Info("Found Valid Kafka Auth Secret")
+		logger.Fatal("error loading eventing-kafka configuration", zap.Error(err))
 	}
 
-	// Extract The Relevant Data From The Kafka Secret
-	kafkaSecretName := kafkaSecret.Name
-	kafkaUsername := string(kafkaSecret.Data[clientconstants.KafkaSecretKeyUsername])
-	kafkaPassword := string(kafkaSecret.Data[clientconstants.KafkaSecretKeyPassword])
-	kafkaSaslType := string(kafkaSecret.Data[clientconstants.KafkaSecretKeySaslType])
-
-	// Create A Kafka Auth Config From Current Credentials (Secret Data Takes Precedence Over ConfigMap)
-	var kafkaAuthCfg *commonclient.KafkaAuthConfig
-	if kafkaUsername != "" {
-		kafkaAuthCfg = &commonclient.KafkaAuthConfig{
-			SASL: &commonclient.KafkaSaslConfig{
-				User:     kafkaUsername,
-				Password: kafkaPassword,
-				SaslType: kafkaSaslType,
-			},
-		}
+	// Extract The Relevant Data From The Kafka Secret And Create A Kafka Auth Config From The
+	// Current Credentials (Secret Data Takes Precedence Over ConfigMap)
+	kafkaAuthCfg, err := config.GetAuthConfigFromKubernetes(ctx, configuration.Kafka.AuthSecretName, configuration.Kafka.AuthSecretNamespace)
+	if err != nil {
+		logger.Fatal("Failed To Load Kafka Auth From Secret", zap.Error(err),
+			zap.String("AuthSecretName", configuration.Kafka.AuthSecretName),
+			zap.String("AuthSecretNamespace", configuration.Kafka.AuthSecretNamespace))
```

Comment:
the namespace and name "flexibility" works for you now?

Reply:
We talked about it and decided that if there's only one secret and only one receiver, there's zero reason to name the receiver after the secret (which was the only reason this option was going to cause trouble).

Although, an upgrade will probably leave an old-named receiver orphaned the way it is; that probably needs its own release note.

Reply:
Or the reconciler could go through and look for secrets that have the old label and delete any deployments/services that match the old hash-name .. that might be a better idea than dumping that responsibility on the admin.  Thoughts?

Reply:
At a minimum we need a release note - it would be bad to have two receiver pods running.

Reply:
After some more discussion we've decided to leave the deployment and service name along (reverted the code back to "use the secret name and hash for the deployment/service names"  Changing that for the sake of appearance would involve writing a new version-upgrade task to move the endpoints of all of the existing KafkaChannel services to point to the newly-renamed Receiver service.  Not impossible by any means but not really worth the bother for what is effectively "just a name change" for this issue.